### PR TITLE
Incorporate ASAN in RVS

### PR DESCRIPTION
**Pull request by manoj-freyr**
Asan enabling in RVS

Signed-off-by: Manoj S K <Manoj.SK@amd.com>

### CODE REVIEW

File: testif.so/CMakeLists.txt [hunk 1]
```diff
@@ -98,7 +98,7 @@ endif()
 set(HIP_HCC_BUILD_FLAGS "${HIP_HCC_BUILD_FLAGS} -DHIP_VERSION_MAJOR=${HIP_VERSION_MAJOR} -DHIP_VERSION_MINOR=${HIP_VERSION_MINOR} -DHIP_VERSION_PATCH=${HIP_VERSION_GITDATE}")
 
 # Add remaining flags
-set(HCC_CXX_FLAGS  "-Xlinker --enable-new-dtags -fno-gpu-rdc --amdgpu-target=gfx803 --amdgpu-target=gfx900 --amdgpu-target=gfx906 --amdgpu-target=gfx908 ")
+set(HCC_CXX_FLAGS  "-Xlinker --enable-new-dtags -fno-gpu-rdc --amdgpu-target=gfx803 --amdgpu-target=gfx900 --amdgpu-target=gfx906 --amdgpu-target=gfx908 -fsanitize=address -shared-libasan ")
```
Comment (ashutom) [inline testif.so/CMakeLists.txt:101]: Is it not possible to change the flags only once & pass it on to all other files?

**Review by ashutom — COMMENTED** [review]
Check if its possible to localize the changes to one one file. 
Bloating it across is violating DRY principle.  Maintaining would be tough going forward.

File: testif.so/CMakeLists.txt (hunk 1 above)
Comment (manoj-freyr) [reply testif.so/CMakeLists.txt]: Incorporated changes as such

**Review by ashutom — COMMENTED** [review]
LGTM!!!

The code has been localized for hot spots.

**Pull request merged**